src/files.rs

```diff
@@ -0,0 +1,173 @@
+use super::ApiResponseOrError;
+use crate::{openai_delete, openai_get, openai_post_multipart, OpenAiError};
+use derive_builder::Builder;
+use reqwest::multipart::{Form, Part};
+use serde::{Deserialize, Serialize};
+use std::path::Path;
+
+#[derive(Deserialize, Serialize, Clone)]
+pub struct File {
+    pub id: String,
+    pub object: String,
+    pub bytes: usize,
+    pub created_at: usize,
+    pub filename: String,
+    pub purpose: String,
+}
+
+#[derive(Deserialize, Serialize, Clone)]
+pub struct DeletedFile {
+    pub id: String,
+    pub object: String,
+    pub deleted: bool,
+}
+
+#[derive(Deserialize, Serialize, Clone)]
+pub struct Files {
+    data: Vec<File>,
+    object: String,
+}
+
+#[derive(Serialize, Builder, Debug, Clone)]
+#[builder(pattern = "owned")]
+#[builder(name = "FileUploadBuilder")]
+#[builder(setter(strip_option, into))]
+pub struct FileUploadRequest {
+    filename: String,
+    purpose: String,
+}
+
+impl File {
+    async fn create(request: &FileUploadRequest) -> ApiResponseOrError<Self> {
+        let purpose = request.purpose.clone();
+        let upload_filepath = Path::new(request.filename.as_str());
+        let upload_filepath = upload_filepath.canonicalize().unwrap();
+        if !upload_filepath.exists() {
+            return Ok(Err(OpenAiError {
+                message: format!("File {} not found", upload_filepath.display()),
+                error_type: "internal".to_string(),
+                param: None,
+                code: None,
+            }));
+        }
+        let simple_name = upload_filepath
+            .file_name()
+            .unwrap()
+            .to_str()
+            .unwrap()
+            .to_string()
+            .clone();
+        let async_file = tokio::fs::File::open(upload_filepath).await.unwrap();
```

Comment:
I would prefer if we got rid of the `unwrap`s here and returned errors instead.
I also plan on flattening the `ApiResponseOrError` type, as nested errors are a bit confusing IMO. But I am happy to merge this in as-is and address the `unwrap`s before the next release.

Reply:
Agreed, the nested errors are proving to be difficult to work with. 
I tried to make it a little better but some tests are showing it's not quite working correctly.

Reply:
Here is a branch with a flattened `ApiResponseOrError`. 

https://github.com/rellfy/openai/compare/master...kcberg:feature/error-handling

If you think this is the right approach for error handling I'll open a PR.

Reply:
I'm happy to go with that. I was thinking to make it enum instead, but that's fine too.